plugins/carbon_black_defense/plugin.spec.yaml

```diff
@@ -523,209 +135,406 @@ types:
-        description: Filter on owner name case insensitive
+      device_name:
+        title: "Device Name"
+        description: "Device name"
         type: string
         required: false
-      ownerNameExact:
-        title: Owners Name Exact
-        description: Same as 'Owners Name' but with case sensitivity
-        type: string
+        example: "CONTOSO\\bruce-win10"
+      device_external_ip:
+        title: "Device External IP"
+        type: "[]string"
+        description: "Device external IP"
         required: false
-      ipAddress:
-        title: IP Address
-        description: Filter on events generated by a device with a given external
-          or internal IP address
-        type: string
-        required: false
-      sha256hash:
+        example: "192.0.2.0"
+      process_hash:
         title: SHA-256 Hash
         description: Filter on events generated by a process with the given SHA-256
           hash. Note that this hash must be lowercase
-        type: string
+        type: "[]string"
         required: false
-      applicationName:
-        title: Application Name
-        description: Filter on events generated by a process with the given application
-          name (for example, googleupdate.exe. Note that this name must be lowercase)
-        type: string
-        required: false
-      eventType:
-        title: Event Type
-        description: Filter on events with a given event type
-        type: string
-        enum:
-        - ''
-        - NETWORK
-        - FILE_CREATE
-        - REGISTRY_ACCESS
-        - SYSTEM_API_CALL
-        - CREATE_PROCESS
-        - DATA_ACCESS
-        - INJECT_CODE
+        example: ["A3DED3871AC66498E406798B0CA33F1B5E94B9CFB60DD4489475B9C19FBFD7F5"]
+      process_name:
+        title: "Process Name"
+        description: "Process name"
+        type: "[]string"
         required: false
-      searchWindow:
-        title: Search Window
-        description: Filter on events generated within a given relative time frame.
-          Note that the default is one day if a searchWindow is not specified. Note
-          that events may not be available past 30 days due to retention policies
-          e.g. 4d, 2w
-        type: string
+        example: "c:\\windows\\system32\\windowspowershell\\v1.0\\modules\\smbshare\\disableunusedsmb1.ps1"
+      enriched_event_type:
+        title: "Enriched Event Type"
+        type: "string"
+        description: "Enriched event type"
         required: false
+        example: "SYSTEM_API_CALL"
+      time_range:
+        window:
+          title: "Time Range"
+          type: "object"
+          description: "Time frame to limit your search to. Integer used for amount of of the following: s=second(s), m=minute(s), h=hour(s), d=day(s), w=week(s), or y=year(s)."
+          required: false
+          enum:
+            -"s"
+            -"m"
+            -"h"
+            -"d"
+            -"w"
+            -"y"
+          example: "-2d"
```

Comment:
The type, enum, and example here aren't compatible. `object` type is for JSON objects, while the `enum` values provided are strings, and the example provided isn't in the enum values. 

I would just change the type to `string`, remove the enum, and leave the example and descriptions as-is 👍🏻

Reply:
Updated with string input